utils/text_formatting.tsx

```diff
@@ -806,5 +820,35 @@ export function handleUnicodeEmoji(text: string, emojiMap: EmojiMap, searchPatte
         // wrap unsupported unicode emoji in span to style as needed
         return `<span class="emoticon emoticon--unicode">${emojiMatch}</span>`;
     });
+
     return output;
 }
+
+// Gets the unicode character code of a character starting at the given index in the string
+// Adapted from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt
+function fixedCharCodeAt(str: string, idx = 0) {
+    // ex. fixedCharCodeAt('\uD800\uDC00', 0); // 65536
+    // ex. fixedCharCodeAt('\uD800\uDC00', 1); // false
+    const code = str.charCodeAt(idx);
+
+    // High surrogate (could change last hex to 0xDB7F to treat high
+    // private surrogates as single characters)
+    if (code >= 0xD800 && code <= 0xDBFF) {
+        const hi = code;
+        const low = str.charCodeAt(idx + 1);
+
+        if (isNaN(low)) {
+            console.log('High surrogate not followed by low surrogate in fixedCharCodeAt()'); // eslint-disable-line
```

Comment:
What's the purpose of logging this?

Reply:
That's just left over from the original code from the MDN. The original version actually threw an error here, but I downgraded it to a console log instead. It's one of those cases that should never happen unless someone is jamming invalid unicode into a post for some reason.